testresults/src/org/labkey/testresults/model/RunDetail.java

```diff
@@ -141,14 +142,24 @@ public void setOs(String os)
         this.os = os;
     }
 
-    public TestMemoryLeakDetail[] getTestmemoryleaks()
+    public TestLeakDetail[] getLeaks()
     {
-        return testmemoryleaks;
+        return leaks;
     }
 
-    public void setTestmemoryleaks(TestMemoryLeakDetail[] testmemoryleaks)
+    public TestMemoryLeakDetail[] getMemoryLeaks()
     {
-        this.testmemoryleaks = testmemoryleaks;
+        return Arrays.stream(leaks).filter(leak -> leak instanceof TestMemoryLeakDetail).toArray(TestMemoryLeakDetail[]::new);
+    }
+
+    public TestHandleLeakDetail[] getHandleLeaks()
```

Comment:
Maybe consider having separate fields for TestMemoryLeakDetail and TestHandleLeakDetail?